Add API for getting PR and Issue data

ajbozarth opened the pull request:
Adds `prs` and `issues` commands that return json objects with pr and issue data.

for ref:
```
$ ./gha prs --help
Get PR data for a given repo and labels

Usage:
  github-activity prs [flags]

Flags:
  -h, --help                help for prs
  -l, --label stringArray   Issue/PR label (default [python])
  -o, --output string       Output Filename
  -r, --repo string         Github org/repo (default "i-am-bee/beeai-framework")

Global Flags:
  -d, --domain string         Github domain (default "github.com")
  -e, --end string            Collect activities up to this date (default "2025-06-06")
  -s, --start string          Collect activities starting on this date (default "2025-05-29")
  -t, --token $GITHUB_TOKEN   Github Personal Access Token (default $GITHUB_TOKEN)
  -u, --user string           Username (default "ajbozarth")
```

ajbozarth committed:
Add api for getting PR data

Signed-off-by: Alex Bozarth <ajbozart@us.ibm.com>

ajbozarth committed:
support quotes around labels

Signed-off-by: Alex Bozarth <ajbozart@us.ibm.com>

ajbozarth committed:
add docs for new flags

Signed-off-by: Alex Bozarth <ajbozart@us.ibm.com>

ajbozarth commented:
@maximilien FYI

psschwei commented:
adding comment from our earlier chat: let's add this functionality as a subcommand rather than a flag. 

You can do that automatically using the [cobra-cli](https://github.com/spf13/cobra-cli) tool or you can just create a file in `cmd/` for the subcommand. See this link for more details: https://github.com/spf13/cobra-cli/blob/main/README.md#add-commands-to-a-project

ajbozarth committed:
Switch from pr flag to prs command

add an output file flag and support for multiple labels

also add a issues command in preparation to add
support for issues, currently just returns prs data

Signed-off-by: Alex Bozarth <ajbozart@us.ibm.com>

ajbozarth commented:
I switched to using commands and added the prs and issues command. Currently issues just runs the prs command, thus why this is still draft.

I also added support to print to file, it might be worth updating the root command to also support this.

My next steps are:
- [x] actually implement the issues command
- [ ] add support for multiple output types (such as json, yaml, and maybe csv)
- [x] cleanup the docs

ajbozarth committed:
Cleanup and add issues support

Signed-off-by: Alex Bozarth <ajbozart@us.ibm.com>

ajbozarth commented:
@psschwei I've cleanup up my draft so it can be merged as is, I don't see myself adding any more features at this point, only some case-by-case query tinkering.

psschwei commented on cmd/issues.go:
```diff
@@ -0,0 +1,25 @@
+}
+
+func init() {
+	issuesCmd.PersistentFlags().StringVarP(&repo, "repo", "r", "i-am-bee/beeai-framework", "Github org/repo")
```
let's not set a default here, since that's something that will be very user-dependent. that'll mean we also need to add a check somewhere to be sure the user passes a value for it

psschwei commented on cmd/prs.go:
```diff
@@ -0,0 +1,25 @@
+}
+
+func init() {
+	prsCmd.PersistentFlags().StringVarP(&repo, "repo", "r", "i-am-bee/beeai-framework", "Github org/repo")
```
same thing here re: default value

psschwei commented on pkg/github/github.go:
```diff
@@ -40,3 +40,81 @@ func GetGithubActivity(domain, startDate, endDate, username, token string) error
+
+}
+
+func GetIssuesData(domain, token, repo, output, enddate, startdate string, labels []string) error {
```
in an ideal world, given the overlap between this function and `GetPRData`, we should probably just have one function and pass in the appropriate github function as an argument... that said, I'm fine to merge this as is (if you wanted to do a follow up to refactor that's fine, if not that's fine too)

ajbozarth replied:
I had planned on doing that, but didn't want to spend the time given I'm moving away from using this next week and felt it wasn't worth the dev time

psschwei commented on cmd/issues.go:
```diff
@@ -0,0 +1,25 @@
+
+func init() {
+	issuesCmd.PersistentFlags().StringVarP(&repo, "repo", "r", "i-am-bee/beeai-framework", "Github org/repo")
+	issuesCmd.PersistentFlags().StringArrayVarP(&labels, "label", "l", []string{"python"}, "Issue/PR label")
```
let's also not set a default on labels, as that will be user-specific too

psschwei commented on pkg/github/helpers.go:
```diff
@@ -127,7 +128,7 @@ type gitHubActivity struct {
 	}
 }
 
-func queryGithubApi(url, query, token string, userActivity *gitHubActivity) error {
+func callGithubApi(url, query, token string) (*http.Response, error) {
```
long-term, it would be better to standardize the flow for issues/prs/activity -- either have them all return the response body and unmarshal, or make the struct parameter more generic so it can support multiple types

for this PR, I don't think it's a blocker

ajbozarth replied:
I had planned on doing that, but didn't want to spend the time given I'm moving away from using this next week and felt it wasn't worth the dev time

psschwei reviewed (commented):
Left some comments, but I also have a general question: is this functionality already handled by the official Github CLI? 

https://cli.github.com/manual/gh_pr_list
https://cli.github.com/manual/gh_issue_list

Actually, looking through their docs, their CLI already does everything this repo does, so it probably makes sense to just use that and archive this repo...

ajbozarth commented:
> Left some comments, but I also have a general question: is this functionality already handled by the official Github CLI?

I took a look too and it does seem that with some tinkering and knowledge we could get all the same data via that CLI, but I would say it's worth merging this PR before archiving this repo since at least we know how to tinker with it better that the gh CLI atm

ajbozarth commented:
@psschwei In ref to the default values, I left those in because I would have to add extra handling for when the user doesn't have them that I didn't want to spend the dev hours on. Given we're considering archiving this repo in luv of gh CLI I don't think it's worth the dev time to solve that. Though I'm willing to change the defaults if you'd like

psschwei commented:
> merging this PR before archiving

I'm good with that. I added my comments before I realized about the gh CLI, so we can skip the ones about refactoring. But for the default values for repo/labels, I would like to drop those. Error handling there should be a simple change, adding something like this in both `GetPRData` and `GetIssuesData`, near where a similar check is done for `token`, would be all that's needed:

```go
if repo == "" {
    return errors.New("no Github repo specified")
}
```

ajbozarth committed:
Address review

remove default values

Signed-off-by: Alex Bozarth <ajbozart@us.ibm.com>

ajbozarth commented:
After deciding to address the defaults issue I realized that cobra might actually have a built in solution, which made it trivial. All review addressed now
